core/src/main/c/share/ooo.h

```diff
@@ -0,0 +1,150 @@
+/*******************************************************************************
+ *     ___                  _   ____  ____
+ *    / _ \ _   _  ___  ___| |_|  _ \| __ )
+ *   | | | | | | |/ _ \/ __| __| | | |  _ \
+ *   | |_| | |_| |  __/\__ \ |_| |_| | |_) |
+ *    \__\_\\__,_|\___||___/\__|____/|____/
+ *
+ *  Copyright (c) 2014-2019 Appsicle
+ *  Copyright (c) 2019-2025 QuestDB
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *  http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ *
+ ******************************************************************************/
+
+
+#ifndef QUESTDB_OOO_H
+#define QUESTDB_OOO_H
+
+constexpr uint8_t DEDUP_INDEX_FORMAT = 1;
+constexpr uint8_t SHUFFLE_INDEX_FORMAT = 2;
+constexpr uint8_t DEDUP_SHUFFLE_INDEX_FORMAT = 3;
+
+template<auto Start, auto End, auto Inc, class F>
+constexpr void constexpr_for(F &&f) {
+    if constexpr (Start < End) {
+        f(std::integral_constant<decltype(Start), Start>());
+        constexpr_for<Start + Inc, End, Inc>(f);
+    }
+}
```

Comment:
Move to `ooo.cpp`, since it's the only place used. Might help with build time slightly.

Reply:
also used in ooo_radix.h